### PR TITLE
Add arm64 docker image (dennisameling work from reverted PR)

### PR DESCRIPTION
**Pull request by rfay**
Right now, this is https://github.com/drud/ddev-images/pull/16, which I mistakenly pulled, and then had to revert. It needed some work on the release creation still, so release creation failed.

### CODE REVIEW

**Comment by dennisameling** [comment]
@rfay let me know if I can be of any help with this 👍

**Comment by rfay** [comment]
Thanks, I think I'm going to get it.  I may have to practice the release build in my own fork. We'll see.

**Comment by rfay** [comment]
A [release build of this branch](https://github.com/rfay/ddev-images/releases/tag/v0.4.0-alpha6) resulted in https://hub.docker.com/layers/drud/ddev-php-base/v0.4.0-alpha6/images/sha256-f6f52b84eb55b936e1848dd051b1e99ba370b5e53b90dacf5fa9ab892d0d186a?context=explore

**Comment by rfay** [comment]
@dennisameling I'd love it if you could take a look at this and see what you think. The release build has now been successful; I'll try a test PR on ddev to make sure it can pass the tests there.

File: Makefile [hunk 1]
```diff
@@ -22,29 +21,51 @@ DEFAULT_IMAGES = ddev-php-base ddev-php-prod
...
-
 build: images
 
 images: $(DEFAULT_IMAGES)
 
-push: images
-	for item in $(DEFAULT_IMAGES); do docker push $(DOCKER_ORG)/$$item:$(VERSION); echo "pushed $(DOCKER_ORG)/$$item"; done
+$(DEFAULT_IMAGES): prep .docker-build-info.txt
+	set -eu -o pipefail; \
+	DOCKER_BUILDKIT=1 docker buildx build --progress=$(PROGRESS) $(BUILDPUSHARG) --platform linux/amd64 --label com.ddev.buildhost=${shell hostname} --target=$@  -t $(DOCKER_ORG)/$@:$(VERSION) $(DOCKER_ARGS) .
```
Comment (dennisameling) [inline Makefile:37]: DOCKER_BUILDKIT=1 is no longer needed with Docker buildx, as [it uses Buildkit by default](https://docs.docker.com/engine/reference/commandline/buildx/). Only found out about this yesterday 😊

File: Makefile [hunk 2]
```diff
@@ -22,29 +21,51 @@ DEFAULT_IMAGES = ddev-php-base ddev-php-prod
...
 	done
 
 version:
 	@echo VERSION:$(VERSION)
 
-buildinfo:
+.docker-build-info.txt:
 	@echo "$(VERSION) $(BUILDINFO)" >.docker-build-info.txt
+
+prep:
```
Comment (dennisameling) [inline Makefile:67]: Assuming you're trying to build from Linux locally? Users of GitHub Desktop (Windows + Mac) don't need this step, that's why I only added it to the CircleCI config previously. But if this is needed to get it to run properly, then by all means 😄

**Review by dennisameling — COMMENTED** [review]
Looks great! Just left two minor comments, apart from those this LGTM and is ready to commit I think 👍 thanks!

File: Makefile (hunk 2 above)
Comment (rfay) [reply Makefile:67]: I'm using macOS edge (stable versions don't have buildx) and it definitely needed this.

File: Makefile (hunk 1 above)
Comment (rfay) [reply Makefile:37]: I knew they were moving that way, but it seems to me the move might not be done yet. They had a bug in macOS with default buildkit on, and I thought they backed away.

https://docs.docker.com/develop/develop-images/build_enhancements/ still shows it as normal to turn it on with DOCKER_BUILDKITE=1, but I'm sure we're going to be wanting to remove that.

**Comment by rfay** [comment]
Pulling with great hope :) Congratulations on all the good work you did on this @dennisameling !

**Pull request merged**